base/sparse/abstractsparse.jl

```diff
@@ -14,3 +16,21 @@ issparse(A::AbstractArray) = false
 issparse(S::AbstractSparseArray) = true
 
 indtype{Tv,Ti}(S::AbstractSparseArray{Tv,Ti}) = Ti
+
+function union!{Tv,Ti,N}(s::AbstractSet, x::AbstractSparseArray{Tv,Ti,N})
+    vals = nonzeros(x)
+    length(vals) == length(x) ? union!(s, vals) : union!(s, [zero(Tv); vals])
+end
+
+# assumes that AbstractSparseArray type stores nonzeros in column major order
```

Comment:
> Should probably only be defining a method that's built on this assumption for concrete sparse types that are known to be column major.

Makes sense. My initial motivation to factor it out was because the implementations were identical. One possible implementation strategy would be to rename the the method to `_unique` in abstractsparse.jl then define `unique` methods for each concrete type that meets the column major requirement (order doesn't matter for `union!`)